### PR TITLE
[LLVM 9] Fixed build errors

### PR DESCRIPTION
**Pull request by smuzaffar**
#### PR description:

Build with LLVM 9.0 fails due to these errors. This PR proposes the changes to make llvm 9 happy

#### PR validation:

locally build with llvm 9.0.1 do not fail any more

### CODE REVIEW

File: Alignment/MillePedeAlignmentAlgorithm/interface/MillePedeFileReader.h [hunk 1]
```diff
@@ -169,7 +169,7 @@ class MillePedeFileReader {
 
 const std::array<std::string, 8> coord_str = {{"X", "Y", "Z", "theta_X", "theta_Y", "theta_Z", "extra_DOF", "none"}};
 inline std::ostream& operator<<(std::ostream& os, const AlignPCLThresholds::coordType& c) {
-  if (c >= AlignPCLThresholds::endOfTypes || c < 0)
+  if (c >= AlignPCLThresholds::endOfTypes || c < AlignPCLThresholds::X)
```
Comment (smuzaffar) [inline Alignment/MillePedeAlignmentAlgorithm/interface/MillePedeFileReader.h]: we can drop `|| c<0` part as it is always `false` 
```
Alignment/MillePedeAlignmentAlgorithm/interface/MillePedeFileReader.h:172:48: error: result of comparison of unsigned enum expression < 0 is always false [-Werror,-Wtautological-unsigned-enum-zero-compare]
```

File: EgammaAnalysis/ElectronTools/interface/EnergyScaleCorrection_class.h [hunk 2]
```diff
@@ -192,7 +192,7 @@ class EnergyScaleCorrection_class {
   void ReadSmearingFromFile(TString filename);  ///< File structure: category constTerm alpha;
 public:
   inline void SetSmearingType(fileFormat_t value) {
-    if (value >= 0 && value <= 1) {
+    if (value <= 1) {
```
Comment (smuzaffar) [inline EgammaAnalysis/ElectronTools/interface/EnergyScaleCorrection_class.h]: `value >= 0` is always true as value is of type fileFormat_t enum.
```
EgammaAnalysis/ElectronTools/interface/EnergyScaleCorrection_class.h:195:15: error: result of comparison of unsigned enum expression >= 0 is always true [-Werror,-Wtautological-unsigned-enum-zero-compare]
```

File: HLTrigger/JetMET/src/HLTJetSortedVBFFilter.cc [hunk 3]
```diff
@@ -162,7 +162,7 @@ bool HLTJetSortedVBFFilter<T>::hltFilter(edm::Event& event,
       nJet++;
       //   		cout << "jetPt=" << jet->pt() << "\tjetEta=" << jet->eta() << "\tjetCSV=" << value << endl;
     }
-    if (b1_idx >= sorted.size() || b1_idx < 0)
+    if (b1_idx >= sorted.size())
```
Comment (smuzaffar) [inline HLTrigger/JetMET/src/HLTJetSortedVBFFilter.cc]: `b1_idx` is `unsigned int` so `b1_idx < 0` is always false
```
HLTrigger/JetMET/src/HLTJetSortedVBFFilter.cc:165:43: error: result of comparison of unsigned expression < 0 is always false [-Werror,-Wtautological-unsigned-zero-compare]
```

File: RecoEgamma/EgammaTools/src/EnergyScaleCorrection.cc [hunk 4]
```diff
@@ -171,7 +171,7 @@ void EnergyScaleCorrection::addSmearing(const std::string& category,
 }
 
 void EnergyScaleCorrection::setSmearingType(FileFormat value) {
-  if (value >= 0 && value <= 1) {
+  if (value <= 1) {
```
Comment (smuzaffar) [inline RecoEgamma/EgammaTools/src/EnergyScaleCorrection.cc]: `value >= 0` is always true as value is of type FileFormat enum.
```
RecoEgamma/EgammaTools/src/EnergyScaleCorrection.cc:174:13: error: result of comparison of unsigned enum expression >= 0 is always true [-Werror,-Wtautological-unsigned-enum-zero-compare]
```

File: RecoTauTag/RecoTau/plugins/DPFIsolation.cc [hunk 5]
```diff
@@ -168,8 +168,6 @@ class DPFIsolation : public deep_tau::DeepTauBase {
 
         if (p.pt() < 0.5)
           continue;
-        if (p.fromPV() < 0)
-          continue;
```
Comment (smuzaffar) [inline RecoTauTag/RecoTau/plugins/DPFIsolation.cc]: `p.fromPV()` return `PVAssoc` enum which is always `>=0` ( https://cmssdt.cern.ch/lxr/source/DataFormats/PatCandidates/interface/PackedCandidate.h#0703 )

```
RecoTauTag/RecoTau/plugins/DPFIsolation.cc:171:24: error: result of comparison of unsigned enum expression < 0 is always false [-Werror,-Wtautological-unsigned-enum-zero-compare]
```

Comment (fwyzard) [reply RecoTauTag/RecoTau/plugins/DPFIsolation.cc]: unless the check was supposed to be `p.fromPV() == 0`  (and `p.fromPV() <= 1` below) ?

**Comment by smuzaffar** [comment]
please test

File: RecoTauTag/RecoTau/plugins/DPFIsolation.cc (hunk 5 above)
Comment (slava77) [reply RecoTauTag/RecoTau/plugins/DPFIsolation.cc]: @ocolegrove @MRD2F @mbluj 
It appears that a check "p.fromPV() < 0" is redundant.
please double-check that we are not hiding some problem in your original code.

File: Alignment/MillePedeAlignmentAlgorithm/interface/MillePedeFileReader.h (hunk 1 above)
Comment (makortel) [reply Alignment/MillePedeAlignmentAlgorithm/interface/MillePedeFileReader.h]: The whole `if` statement seems to be redundant (unless something casts an integer into the enumeration without bounds checking)
https://github.com/cms-sw/cmssw/blob/dcdf62c803c6e74bdebb31cfbae0021aeaaa5a57/CondFormats/PCLConfig/interface/AlignPCLThresholds.h#L14

File: RecoTauTag/RecoTau/plugins/DPFIsolation.cc (hunk 5 above)
Comment (mbluj) [reply RecoTauTag/RecoTau/plugins/DPFIsolation.cc]: I guess it is fine the remove the `p.fromPV() < 0` redundant check.

**Comment by Martin-Grunewald** [comment]
+1

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/28810	dcdf62c 
- mostly technical, as described
- also, one somewhat questionable case was confirmed in https://github.com/cms-sw/cmssw/pull/28810#discussion_r372846716

**Comment by christopheralanwest** [comment]
+1

**Comment by silviodonato** [comment]
merge
@santocch please have a look

**Comment by santocch** [comment]
+1

**Pull request merged**